lib/temporal/connection/grpc.rb

```diff
@@ -31,14 +31,15 @@ def initialize(host, port, identity)
         @poll_request = nil
       end
 
-      def register_namespace(name:, description: nil, global: false, retention_period: 10)
+      def register_namespace(name:, description: nil, global: false, retention_period: 10, namespace_data: nil)
         request = Temporal::Api::WorkflowService::V1::RegisterNamespaceRequest.new(
           namespace: name,
           description: description,
           is_global_namespace: global,
           workflow_execution_retention_period: Google::Protobuf::Duration.new(
             seconds: retention_period * 24 * 60 * 60
```

Comment:
This seems like a latent bug. Should wrap this whole thing and call `to_i` on it to convert to integer.

Reply:
Sorry wrap this whole thing in what?

Reply:
Heh, it's a Ruby thing. If you do `(3.14159).to_i` it returns `3`. It just converts floats to ints, that's all.
```suggestion
            seconds: (retention_period * 24 * 60 * 60).to_i
```

Reply:
ah thank you!